cfme/tests/v2v/test_v2v_migrations.py

```diff
@@ -441,3 +299,283 @@ def test_conversion_host_tags(appliance, v2v_providers):
     host.remove_tag(tag2)
 
     host.remove_tags(host.get_tags())
+
+
+@pytest.mark.parametrize('conversion_tags, form_data_vm_obj_single_datastore', [['SSH',
+    ['nfs', 'nfs', rhel7_minimal]]], indirect=True)
+def test_single_vm_migration_with_ssh(request, appliance, v2v_providers, host_creds,
+                                    conversion_tags,
+                                    form_data_vm_obj_single_datastore):
+    infrastructure_mapping_collection = appliance.collections.v2v_mappings
+
+    mapping = infrastructure_mapping_collection.create(form_data_vm_obj_single_datastore[0])
+
+    @request.addfinalizer
+    def _cleanup():
+        infrastructure_mapping_collection.delete(mapping)
+
+    migration_plan_collection = appliance.collections.v2v_plans
+
+    migration_plan = migration_plan_collection.create(
+        name="plan_{}".format(fauxfactory.gen_alphanumeric()), description="desc_{}"
+        .format(fauxfactory.gen_alphanumeric()), infra_map=mapping.name,
+        vm_list=form_data_vm_obj_single_datastore[1], start_migration=True)
+
+    # explicit wait for spinner of in-progress status card
+    view = appliance.browser.create_view(navigator.get_class(migration_plan_collection, 'All').VIEW)
+    wait_for(func=view.progress_card.is_plan_started, func_args=[migration_plan.name],
+        message="migration plan is starting, be patient please", delay=5, num_sec=150,
+        handle_exception=True)
+
+    # wait until plan is in progress
+    wait_for(func=view.get_plan_status, func_args=[migration_plan.name],
+        message="migration plan is in progress, be patient please",
+        delay=5, num_sec=3600)
+
+    view.migr_dropdown.item_select("Completed Plans")
+    view.wait_displayed()
+    logger.info("For plan %s, migration status after completion: %s, total time elapsed: %s",
+        migration_plan.name, view.migration_plans_completed_list.get_vm_count_in_plan(
+            migration_plan.name), view.migration_plans_completed_list.get_clock(
+            migration_plan.name))
+    assert view.migration_plans_completed_list.is_plan_succeeded(migration_plan.name)
+
+
+@pytest.mark.parametrize('form_data_vm_obj_single_datastore', [['nfs', 'nfs', rhel7_minimal]],
+    indirect=True)
+@pytest.mark.parametrize('power_state', ['RUNNING', 'STOPPED'])
+def test_single_vm_migration_power_state_tags_retirement(request, appliance, v2v_providers,
+                                    host_creds, conversion_tags,
+                                    form_data_vm_obj_single_datastore,
+                                    power_state):
+    # Test VM migration power state and tags are preserved
+    vm = form_data_vm_obj_single_datastore[1][0]
```

Comment:
As @mshriver stated above, some clarity on the indexing here and below would be helpful.